### DISCUSSION TITLE
Amazing tool! Looks promising! But…

### DISCUSSION THREAD

**Discussion by ivan-avalos** [dirkpetersen/froster#13] [General]
Hey! Minutes ago, I was looking for cheap backup solutions for my self-hosted system, and came across S3 Glacier. Doing a quick GitHub search, looking for some nice freely-licensed client optimized for backups, I found Froster. It has 0 stars, it was published six months ago, and apparently almost no one has ever heard about it; however, it looks too good to be true. It even has a nice logo and a friendly and comprehensive README. Good job!

My question is, what's Froster's current status? Is it well-tested, “production ready” and something I can rely on? I don't expect perfection, of course, but when it comes to backups, it is reasonable to want to make sure that, as long as I use it properly, it will never mess up my data.

**Edit:** I just read in the README that 50% of the code was AI generated, which doesn't inspire confidence. No matter how good AI assistants are getting, generative coding is still a relatively new technology that has yet to prove its reliability. Now, even more so, I'm interested in knowing how extensively has this tool been used and tested.

**Comment by dirkpetersen** [comment]
Good questions, thanks. Glacier is currently used by multiple research labs and it has been doing a reliable job. I have put it together because researchers were spending too much time managing their data archiving processes. It should also work for backups but if you are not restoring stuff frequently and always know where the data is you want to backup perhaps you just use Rclone? Froster uses Rclone in the background and you can use Rclone to push directly to Glacier (you just cannot get it back with just Rclone).

**Comment by dirkpetersen** [comment]
As to AI generated code, of course you always have to review every AI generated thing, be it code or text or anything. But reviewing code is a lot faster than writing it .... plus ChatGPT writes better than most people and Github copilot is also nice

**Comment by macheitor** [comment]
> Hey! Minutes ago, I was looking for cheap backup solutions for my self-hosted system, and came across S3 Glacier. Doing a quick GitHub search, looking for some nice freely-licensed client optimized for backups, I found Froster. It has 0 stars, it was published six months ago, and apparently almost no one has ever heard about it; however, it looks too good to be true. It even has a nice logo and a friendly and comprehensive README. Good job!
> 
> My question is, what's Froster's current status? Is it well-tested, “production ready” and something I can rely on? I don't expect perfection, of course, but when it comes to backups, it is reasonable to want to make sure that, as long as I use it properly, it will never mess up my data.
> 
> **Edit:** I just read in the README that 50% of the code was AI generated, which doesn't inspire confidence. No matter how good AI assistants are getting, generative coding is still a relatively new technology that has yet to prove its reliability. Now, even more so, I'm interested in knowing how extensively has this tool been used and tested.

Hi @ivan-avalos 

My name is Victor, and I have been working on the Froster project for the last three months. Now, it is safe to say that all of Froster's code has been rewritten/revised by a Software Engineer with many years of experience in programming. Froster is no longer an AI-generated tool.

I encourage you to try the new Froster and let it help you with your filesystem management. 

Best regards
Victor
